dashboard/src/pages/[workspaceSlug]/[appSlug]/users/index.tsx

```diff
@@ -1,38 +1,379 @@
-import { LoadingScreen } from '@/components/common/LoadingScreen';
+import { useDialog } from '@/components/common/DialogProvider';
+import Pagination from '@/components/common/Pagination';
 import Container from '@/components/layout/Container';
 import ProjectLayout from '@/components/layout/ProjectLayout';
-import UsersList from '@/components/users/UsersList';
+import type { EditUserFormValues } from '@/components/users/EditUserForm';
+import UsersBody from '@/components/users/UsersBody';
 import { useCurrentWorkspaceAndApplication } from '@/hooks/useCurrentWorkspaceAndApplication';
+import { useRemoteApplicationGQLClient } from '@/hooks/useRemoteApplicationGQLClient';
+import Button from '@/ui/v2/Button';
+import Input from '@/ui/v2/Input';
+import Text from '@/ui/v2/Text';
+import PlusIcon from '@/ui/v2/icons/PlusIcon';
+import UserIcon from '@/ui/v2/icons/UserIcon';
+import type { RemoteAppGetUsersQuery } from '@/utils/__generated__/graphql';
+import {
+  useGetRolesQuery,
+  useRemoteAppDeleteUserMutation,
+  useRemoteAppGetUsersQuery,
+  useUpdateRemoteAppUserMutation,
+} from '@/utils/__generated__/graphql';
 import { generateAppServiceUrl } from '@/utils/helpers';
-import { NhostApolloProvider } from '@nhost/react-apollo';
-import type { ReactElement } from 'react';
+import { toastStyleProps } from '@/utils/settings/settingsConstants';
+import { SearchIcon } from '@heroicons/react/solid';
+import axios from 'axios';
+import debounce from 'lodash.debounce';
+import type { ChangeEvent, ReactElement } from 'react';
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { toast } from 'react-hot-toast';
+
+export type RemoteAppUser = Exclude<
+  RemoteAppGetUsersQuery['users'][0],
+  '__typename'
+>;
 
 export default function UsersPage() {
+  const { openDialog, openAlertDialog, openDrawer, closeDrawer, closeDialog } =
+    useDialog();
+  const remoteProjectGQLClient = useRemoteApplicationGQLClient();
+  const [searchString, setSearchString] = useState<string>('');
+  const [currentPage, setCurrentPage] = useState(1);
+  const [deleteUser] = useRemoteAppDeleteUserMutation({
+    client: remoteProjectGQLClient,
+  });
+  const [updateUser] = useUpdateRemoteAppUserMutation({
+    client: remoteProjectGQLClient,
+  });
+  const limit = useRef(25);
+  const [nrOfPages, setNrOfPages] = useState(1);
```

Comment:
I think we should also support storing the current page as a query parameter, just like we do on the Storage page.

Reply:
Yes, the current page should be a `?page=<page>` query parameter.

Reply:
I'll add this